doorkeeper-openid_connect.gemspec

```diff
@@ -24,7 +24,7 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 2.4'
 
-  spec.add_runtime_dependency 'doorkeeper', '>= 5.2', '< 5.5'
+  spec.add_runtime_dependency 'doorkeeper', '>= 5.5', '< 5.6'
```

Comment:
Do we need such strict dependency ? This extension currently supports previous versions of Doorkeeper (check  lib/doorkeeper/openid_connect.rb#63 - there is a check `if defined?(::Doorkeeper::GrantFlow)`)

Reply:
@linhdangduy could you please address this? :)

Reply:
> Do we need such strict dependency ? This extension currently supports previous versions of Doorkeeper (check lib/doorkeeper/openid_connect.rb#63 - there is a check if defined?(::Doorkeeper::GrantFlow))

Sorry for the delay.

Yes, we need.

The change that I made was adding [response_mode_matches](https://github.com/doorkeeper-gem/doorkeeper/pull/1438/files#diff-fbf6e8847bf493e8a4c566038f5bec7700e78737b1af85723ddf5790a5856375R16) attribute to `::Doorkeeper::GrantFlow`.  That change is released in v5.5.
So the doorkeeper version below v5.5, with `::Doorkeeper::GrantFlow` defined, does not have that attribute, causing the error we met.

Reply:
OK, let's call @toupeira  also for a review :)

Reply:
@nbulaj @linhdangduy thanks for the ping!

I'm fine with requiring Doorkeeper 5.5, as the two gems are very tightly coupled anyway :+1:

I see we can remove the code at https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/805b7ca9c42a011e357b2d6daac88c45e7f2172a/lib/doorkeeper/openid_connect.rb#L85-L87 and the file https://github.com/doorkeeper-gem/doorkeeper-openid_connect/blob/805b7ca9c42a011e357b2d6daac88c45e7f2172a/lib/doorkeeper/openid_connect/response_types_config.rb, @linhdangduy do you want to do that in this PR while we're at it? Otherwise I'd also be happy to do it in a follow-up.

Reply:
@toupeira Thanks for the review. I will take a look today.

Reply:
https://github.com/doorkeeper-gem/doorkeeper-openid_connect/pull/138/commits/99ef25ed9b6949c64d5aff08e7b29e6349cb393e

I removed that unnecessary part.